cpp/oneapi/dal/test/engine/dataframe_common.cpp

```diff
@@ -219,6 +219,30 @@ class dataframe_builder_action_allocate : public dataframe_builder_action {
     std::int64_t column_count_;
 };
 
+class dataframe_builder_action_set : public dataframe_builder_action {
+public:
+    explicit dataframe_builder_action_set(const array<float>& data,
+                                          std::int64_t row_count,
+                                          std::int64_t column_count)
+            : array_(data),
+              row_count_(row_count),
+              column_count_(column_count) {}
+
+    std::string get_opcode() const override {
+        return fmt::format("set({},{})", row_count_, column_count_);
```

Comment:
```suggestion
        return fmt::format("read_external_dataset({})", dataset_relative_path);
```

Opcode is used as a cache key. Caching is important to speedup dataset generation, if entry with the same key is found, the already loaded dataset is used. You don't need to care about caching, it's done as part of dataframe infrastructure. The only thing you have to do is to create unique `opcode`.